File: Java/Esercizi/Forno/ProdottoForno.java
public abstract class ProdottoForno{
  protected String codice;
  protected String nome;
  protected int giorniDurata;

  public abstract float prezzoFinale();

	public String toString(){
    	return  "Codice: " + codice + "\nnome: " + nome + "\nGiorniDurata: " + giorniDurata;
  	}


  public ProdottoForno(){
    this.codice = "";
    this.nome = "";
    this.giorniDurata = 0;
  }

  public ProdottoForno(String codice, String nome, int giorniDurata){
    this.codice = codice;
    this.nome = nome;
    this.giorniDurata = giorniDurata;
  }
  
  public ProdottoForno(ProdottoForno c){
    this.codice = c.codice;
    this.nome = c.nome;
    this.giorniDurata = c.giorniDurata;
  }


	public String getCodice(){
		return codice;
	}

	public String getNome(){
		return nome;
	}

	public int getGiorniDurata(){
		return giorniDurata;
	}

	
	public void setCodice(String c){
		codice = c;
	}

	public void setNome(String s){
		nome = s;
	}

	public void setGiorniDurata(int g){
		giorniDurata = g;
	}
}

File: forno/prodottoForno.cpp
#include "prodottoForno.h"
#include <string>

using namespace std;

// coststruttore di default
ProdottoForno::ProdottoForno(){
    scadenza=0; 
    codice=""; 
    nome="";  
}

// costruttore con parametri
ProdottoForno::ProdottoForno(string p_codice, string p_nome, int p_scadenza){
  scadenza=p_scadenza; 
  codice=p_codice; 
  nome=p_nome;  
}

int ProdottoForno::getScadenza()
{
  return scadenza; 
}

string ProdottoForno::getNome()
{
  return nome; 
}

string ProdottoForno::getCodice(){
    return codice; 
}

void ProdottoForno::setScadenza(int p_scadenza){
    scadenza=p_scadenza; 
}

void ProdottoForno::setNome(string p_nome){
  nome=p_nome; 
}

void ProdottoForno::setCodice(string p_codice){
  codice=p_codice; 
}